tests/update_files.py

```diff
@@ -28,126 +29,25 @@
-
-    # if no output was found in the updated outputs directory, that's
-    # likely to be a problem
-    if not dir_comparison.left_list:
-        sys.stderr.write("\nError: no updated outputs found for \"{}\".\n".format(source))
-        sys.exit(1)
-
-    # first delete the files that only exist in the existing output directory
-    # since those are likely old files from old versions that we do not need
-    existing_output_only_files = dir_comparison.right_only
-    for file in existing_output_only_files:
-        remove(existing_output_path / file)
-
-    # Next find all the NEW files in the updated outputs but do not include
-    # the config JSON files or the OLS summary files or
-    # the evaluation/prediction/model files for SKLL models
-    new_files = dir_comparison.left_only
-    new_files = [f for f in new_files if not f.endswith('_rsmtool.json') and not f.endswith('_rsmeval.json') and not f.endswith('_ols_summary.txt')]
-    if skll:
-        new_files = [f for f in new_files if not is_skll_excluded_file(f)]
-
-    # next we get the files that have changed and try to figure out if they
-    # have actually changed beyond a tolerance level that we care about for
-    # tests. To do this, we run the same function that we use when comparing
-    # the files in the actual test. However, for non-tabular files, we just
-    # assume that they have really changed since we have no easy way to compare.
-    changed_files = dir_comparison.diff_files
-    really_changed_files = []
-    for changed_file in changed_files:
-        include_file = True
-        updated_output_filepath = updated_output_path / changed_file
-        existing_output_filepath = existing_output_path / changed_file
-        file_format = updated_output_filepath.suffix.lstrip('.')
-        if file_format in ['csv', 'tsv', 'xlsx']:
-            try:
-                check_file_output(str(updated_output_filepath),
-                                  str(existing_output_filepath),
-                                  file_format=file_format)
-            except AssertionError:
-                pass
-            else:
-                include_file = False
-
-        if include_file:
-            really_changed_files.append(changed_file)
-
-    # Copy over the new files as well as the really changed files
-    new_or_changed_files = new_files + really_changed_files
-    for file in new_or_changed_files:
-        copyfile(updated_output_path / file, existing_output_path / file)
-
-    # return lists of files we deleted and updated
-    return sorted(existing_output_only_files), sorted(new_or_changed_files)
+from rsmtool.test_utils import FileUpdater
 
 
 def main():
     # set up an argument parser
     parser = argparse.ArgumentParser(prog='update_test_files.py')
-    parser.add_argument('outputs_dir', help="The path to the directory containing the updated test outputs")
+    parser.add_argument('--tests',
```

Comment:
Potentially a nitpick: these are treated as optional arguments when you call the script so you get a Traceback error (Path is None) if you forget to specify them.

Reply:
I can just add `required=True` to these.